### DISCUSSION TITLE
ESP32 GPIO pin hold RTC and deepsleep

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#9880] [Hardware & Peripherals] [answered]
I can hold a RTC-capable port during deepsleep() but if I use esp32.wake_on_ext0() when the ESP32 reboots it changes to low for about 1 sec, until told to go high again.

A 10Kpull-up brings the voltage up to about 2.0V, which might work.

Is there a better way to hold a pin during reboot?

Thanks

**Comment by karfas** [comment, marked as answer]
You might want to check the data sheet, but I don't think you can reliable control the state of a GPIO when the ESP32 is not running or during boot.
When the state of the specific GPIO is critical for your application and must not be low for any reason, you will need external hardware (your pull-up, a transistor, latches, whatever) to enforce this. What should happen when the ESP32 looses it's power supply and the rest of your circuit is still powered ?

**Reply by davefes** [reply]
Thank you for your comments.
